src/models/Chemistry.ts

```diff
@@ -259,48 +261,91 @@ function typesMatch(compound1: (Element | Bracket)[], compound2: (Element | Brac
     if (isElement(test) && isElement(target)) {
-        response.isEqual = response.isEqual &&
-            test.value === target.value &&
-            test.coeff === target.coeff;
-        response.sameCoefficient = response.sameCoefficient && test.coeff === target.coeff;
+        if (!response.allowPermutations) {
+            response.isEqual = response.isEqual &&
+                test.value === target.value &&
+                test.coeff === target.coeff;
+            response.sameCoefficient = response.sameCoefficient && test.coeff === target.coeff;
+        }
 
-        if (response.atomCount) {
-            response.atomCount[test.value] = (response.atomCount[test.value] ?? 0) + test.coeff;
+        if (test.bracketed) {
+            if (response.bracketAtomCount) {
+                response.bracketAtomCount[test.value] = (response.bracketAtomCount[test.value] ?? 0) + test.coeff;
+            } else {
+                response.bracketAtomCount = {} as Record<ChemicalSymbol, number | undefined>;
+                response.bracketAtomCount[test.value] = test.coeff;
+            }
         } else {
-            response.atomCount = {} as Record<ChemicalSymbol, number | undefined>;
-            response.atomCount[test.value] = test.coeff;
+            if (response.termAtomCount) {
+                response.termAtomCount[test.value] = (response.termAtomCount[test.value] ?? 0) + test.coeff;
+            } else {
+                response.termAtomCount = {} as Record<ChemicalSymbol, number | undefined>;
+                response.termAtomCount[test.value] = test.coeff;
+            }
         }
         return response;
     }
     else if (isBracket(test) && isBracket(target)) {
+        test.compound.bracketed = true;
+        target.compound.bracketed = true;
         const newResponse = checkNodesEqual(test.compound, target.compound, response);
 
         newResponse.sameCoefficient = newResponse.sameCoefficient && test.coeff === target.coeff;
         newResponse.isEqual = newResponse.isEqual && test.coeff === target.coeff;
 
-        if (newResponse.atomCount) {
-            for (const [key, value] of Object.entries(newResponse.atomCount)) {
-                newResponse.atomCount[key as ChemicalSymbol] = (value ?? 0) * test.coeff;
+        if (newResponse.bracketAtomCount) {
+            for (const [key, value] of Object.entries(newResponse.bracketAtomCount)) {
+                if (newResponse.termAtomCount) {
+                    newResponse.termAtomCount[key as ChemicalSymbol] = (newResponse.termAtomCount[key as ChemicalSymbol] ?? 0) + (value ?? 0) * test.coeff;
+                }
+                else {
+                    newResponse.termAtomCount = {} as Record<ChemicalSymbol, number | undefined>;
+                    newResponse.termAtomCount[key as ChemicalSymbol] = (value ?? 0) * test.coeff;
+                }
+                newResponse.bracketAtomCount = {} as Record<ChemicalSymbol, number | undefined>;
```

Comment:
Would it not be best to have this outside of the for loop so that it doesn't keep resetting it? I know entries creates a new object, but this feels very coutner-intuitive on first read.